test/integration/rolesAPITests.coffee

```diff
@@ -431,6 +431,31 @@ describe 'API Integration Tests', ->
           .expect(403)
           .end (err, res) -> done err
 
+      it 'should add a role for clients', (done) ->
+        request('https://localhost:8080')
+          .post('/roles')
+          .set('auth-username', testUtils.rootUser.email)
+          .set('auth-ts', authDetails.authTS)
+          .set('auth-salt', authDetails.authSalt)
+          .set('auth-token', authDetails.authToken)
+          .send
+            name: 'role4'
+            clients: [
+                _id: "#{client1._id}"
+              ,
+                _id: "#{client2._id}"
+            ]
+          .expect(201)
+          .end (err, res) ->
+            return done err if err
+            Client.find roles: $in: ['role4'], (err, clients) ->
+              return done err if err
+              clients.length.should.be.exactly 2
+              mapId = (arr) -> (arr.map (a) -> "#{a._id}")
+              mapId(clients).should.containEql "#{client1._id}"
+              mapId(clients).should.containEql "#{client2._id}"
+              done()
```

Comment:
Already done for update, see `should add a role for clients` below. Will add for query.